The bounds of each _WKTextRun should contain the text range (instead of the entire node)

whsieh opened the pull request:
#### 6d776ad376d9e10dd25da4baafc8b558df896a9f
<pre>
The bounds of each _WKTextRun should contain the text range (instead of the entire node)
<a href="https://bugs.webkit.org/show_bug.cgi?id=297308">https://bugs.webkit.org/show_bug.cgi?id=297308</a>
<a href="https://rdar.apple.com/158187835">rdar://158187835</a>

Reviewed by Abrar Rahman Protyasha.

Currently, each visible run of text extracted through this SPI reports its containing text node&apos;s
bounds as the `rectInWebView`. However, this is inconsistent with the text run itself, which is
limited to a single line. Fix this inconsistency by instead reporting the bounds of the text rects
corresponding to each range surfaced by `TextIterator`.

* Source/WebCore/page/text-extraction/TextExtraction.cpp:
(WebCore::TextExtraction::extractAllTextAndRectsRecursive):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewGetContents.mm:
(TestWebKitAPI::TEST(WKWebView, RequestAllTextRunsWithSubframes)):

Rebaseline an existing API test to exercise this fix.

Canonical link: <a href="https://commits.webkit.org/298609@main">https://commits.webkit.org/298609@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a05aa97424488132f82f7a8bf2b325c0a7562494

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116043 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35704 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26245 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122099 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/66591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1c13dadf-2327-436e-962c-1db36806c754) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117932 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36398 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44292 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88166 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/66591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7f4731eb-6e53-43b4-ad15-3e9c56d3bcbf) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118991 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29033 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104125 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68565 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28153 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22234 "Found 1 new test failure: imported/w3c/web-platform-tests/event-timing/only-observe-firstInput.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65781 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98422 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22370 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125249 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42937 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32227 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/96900 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43302 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100315 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96684 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24602 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41952 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19829 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/38883 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42824 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48416 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42291 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45625 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43995 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a05aa97424488132f82f7a8bf2b325c0a7562494. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49307

whsieh commented:
Thanks for the review!

(`mac-wk1` test failure is unrelated)

whsieh committed:
The bounds of each _WKTextRun should contain the text range (instead of the entire node)
https://bugs.webkit.org/show_bug.cgi?id=297308
rdar://158187835

Reviewed by Abrar Rahman Protyasha.

Currently, each visible run of text extracted through this SPI reports its containing text node's
bounds as the `rectInWebView`. However, this is inconsistent with the text run itself, which is
limited to a single line. Fix this inconsistency by instead reporting the bounds of the text rects
corresponding to each range surfaced by `TextIterator`.

* Source/WebCore/page/text-extraction/TextExtraction.cpp:
(WebCore::TextExtraction::extractAllTextAndRectsRecursive):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewGetContents.mm:
(TestWebKitAPI::TEST(WKWebView, RequestAllTextRunsWithSubframes)):

Rebaseline an existing API test to exercise this fix.

Canonical link: https://commits.webkit.org/298609@main

webkit-commit-queue commented:
Committed 298609@main (6d776ad376d9): <https://commits.webkit.org/298609@main>

Reviewed commits have been landed. Closing PR #49307 and removing active labels.
